tex2pdf-tools/tex2pdf_tools/preflight_parser/__init__.py

```diff
@@ -1155,13 +1219,81 @@ def deal_with_bibliographies(
             # TODO, maybe remove issues with missing .bib files?
             continue
         # toplevel filename .bbl is missing -> require .bib to be available,
-        top_node = nodes[tl_f]
-        all_bib = top_node.recursive_collect_files(FileType.bib)
+        all_bib = node.recursive_collect_files(FileType.bib)
         if all_bib:
             # TODO detect biber usage!
             tl_n.process.bibliography = BibProcessSpec(processor=BibCompiler.unknown, pre_generated=False)
 
 
+def deal_with_indices(rundir: str, toplevel_files: dict[str, ToplevelFile], nodes: dict[str, ParsedTeXFile]) -> None:
+    """Check for inclusion of idx files and presence .ind files."""
+    for tl_f, tl_n in toplevel_files.items():
+        node = nodes[tl_f]
+        # first collect the *defined* indices which contain the tag name:
+        #          "used_ind_files" : [
+        #             "<MAIN>.<tag1>",
+        #             "<MAIN>.<tag2>"
+        #          ],
+        #          "used_idx_files": [
+        #             "<MAIN>.<tag1>.<idx_ext>.<ind_ext>",
+        #             ....
+        #          ]
+        all_ind = [fn[8:-1] for fn in node.recursive_collect_files(FileType.ind)]
+        all_idx = [fn[7:] for fn in node.recursive_collect_files(FileType.idx)]
```

Comment:
Just to be sure, is the idea that the `8:-1` also unwraps a leading/trailing pair of `<>` characters? And `7:` leaves them in for the idx case, since there are more components than just 1?

Reply:
Yes, exactly. Dealing with these \jobname.ext files is a bit a pain because we don't know at parse time what the \jobname will be (main file name).